ext/js/dictionary/dictionary-database.js

```diff
@@ -128,8 +151,30 @@ export class DictionaryDatabase {
                         },
                     },
                 },
-            ]),
+            ]);
+        await this._db.open(
+            this._dbName,
+            60,
+            upgrade,
         );
+
+        // when we are not a worker ourselves, create a worker which is basically just a wrapper around this class, which we can use to offload some functions to
```

Comment:
Unfortunately, not exactly. On Chrome, we expect this to be running in the offscreen context, and in Firefox we expect it to be in the background page context. In theory it could also run in the SW context in Chrome though we don't do that currently since SW is short-lived and we want to keep the DB open longer. These contexts are neither threads or processes, as threads and processes are implementation details of the browser. (e.g., in Chrome each extension runs in its own process, and I'm not sure if SW and offscreen are in different processes, while on Firefox all extensions run in a single process, but any extension resources which are loaded as a child of a non-extension resource seems to be in the normal webpage process).

So "not a worker" is the most technically accurate way to capture all the possible situations here.